src/runtime/str.cpp

```diff
@@ -46,16 +46,33 @@ extern "C" PyObject* string_splitlines(PyStringObject* self, PyObject* args) noe
 
 namespace pyston {
 
-BoxedString::BoxedString(const char* s, size_t n) : s(s, n) {
-    gc::registerGCManagedBytes(this->s.size());
+BoxedString::BoxedString(const char* s, size_t n) : s(storage(), n) {
+    if (s) {
+        memmove(storage(), s, n);
+        storage()[n] = 0;
+    } else {
+        memset(storage(), 0, n + 1);
+    }
+    gc::registerGCManagedBytes(cls->tp_basicsize + this->s.size() + 1);
```

Comment:
yup, the various arena `alloc` functions already register.  so with these lines we double count string memory.